src/main/java/redis/clients/jedis/ClientSetInfoConfig.java

```diff
@@ -1,18 +1,70 @@
 package redis.clients.jedis;
 
-/**
- * This interface is to modify the behavior of internally executing CLIENT SETINFO command.
- */
-public interface ClientSetInfoConfig {
+import java.util.Arrays;
+import java.util.HashSet;
+import redis.clients.jedis.exceptions.JedisValidationException;
 
-  default boolean isDisabled() {
-    return false;
+public final class ClientSetInfoConfig {
+
+  private final boolean disabled;
+
+  private final String libNameSuffix;
+
+  public ClientSetInfoConfig() {
+    this(false, null);
+  }
+
+  public ClientSetInfoConfig(boolean disabled) {
+    this(disabled, null);
+  }
+
+  /**
+   * @param libNameSuffix must not have braces ({@code ()[]{}}) and spaces will be replaced with hyphens
+   */
+  public ClientSetInfoConfig(String libNameSuffix) {
+    this(false, libNameSuffix);
+  }
+
+  private ClientSetInfoConfig(boolean disabled, String libNameSuffix) {
+    this.disabled = disabled;
+    this.libNameSuffix = validateLibNameSuffix(libNameSuffix);
   }
 
+  private static final HashSet<Character> BRACES = new HashSet<>(Arrays.asList('(', ')', '[', ']', '{', '}'));
+
+  private static String validateLibNameSuffix(String suffix) {
+    if (suffix == null || suffix.trim().isEmpty()) {
+      return null;
+    }
+
+    for (int i = 0; i < suffix.length(); i++) {
+      char c = suffix.charAt(i);
+      if (c < ' ' || c > '~' || BRACES.contains(c)) {
```

Comment:
IMHO why not use java.util.regex.Matcher and java.util.regex.Pattern; Rather than roll your own via a for loop, regular expressions are pretty performant. Is this faster?

Reply:
https://github.com/redis/jedis/pull/3458#discussion_r1218063512 
Please check full discussion.

Reply:
Awesome - hadn't seen that. Love it.